bivrost-solidity-types/src/main/kotlin/pm/gnosis/model/SolidityBase.kt

```diff
@@ -376,11 +357,11 @@ object SolidityBase {
     fun decodeString(source: PartitionData) =
             decodeBytes(source).toString(Charset.forName("UTF-8"))
 
+    @Deprecated("Deprecated for decodeList")
     fun <T : Any> decodeArray(data: String, itemDecoder: (String) -> T): List<T> {
-        val params = partitionData(data)
-        val contentSize = BigDecimal(BigInteger(params[0])).intValueExact()
-        if (contentSize != params.size - 1) throw IllegalArgumentException("Number of items is different from the actual array size")
+        val params = PartitionData.of(data)
+        val contentSize = BigInteger(params.consume()).intValueExact()
         if (contentSize == 0) return emptyList()
-        return (1 until params.size).map { itemDecoder.invoke(params[it]) }.toList()
+        return (0 until contentSize).map { itemDecoder.invoke(params.consume()) }.toList()
```

Comment:
Do we need the `invoke()` here?

Reply:
I think having
```kotlin
return (0 until contentSize).map { itemDecoder.invoke(params.consume()) }
```
is equivalent.

`.map { }` handles empty lists and it also returns a list so `.toList()` is not necessary.